src/components/step-flow/step-flow.pw.tsx

```diff
@@ -0,0 +1,275 @@
+          `${category}. ${title}. Step ${currentStep} of ${totalSteps}.`
+        );
+      } else {
+        await expect(stepFlowVisuallyHiddenTitleText(page)).toHaveText(
+          `${category}. ${title}. Step ${totalSteps} of ${totalSteps}.`
+        );
+      }
+    });
+  });
+
+  ([1, 2, 3, 4, 5, 6, 7, 8] as Steps[]).forEach((totalSteps) => {
+    test(`should render the correct amount of progress indicators when the 'totalSteps' prop is passed as ${totalSteps}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <StepFlowComponent totalSteps={totalSteps} showProgressIndicator />
+      );
+
+      expect(await stepFlowProgressIndicator(page).count()).toEqual(totalSteps);
+    });
+  });
+
+  test("should render the correct element when 'showCloseIcon' is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<StepFlowComponent showCloseIcon />);
+
+    await expect(stepFlowDismissIcon(page)).toBeVisible();
+  });
+
+  test("background colour token should be correct when 'showCloseIcon' is true and the indicator is in progress", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <StepFlowComponent currentStep={1} totalSteps={1} showProgressIndicator />
+    );
+
+    const backgroundColorToken = await getDesignTokensByCssProperty(
+      page,
+      stepFlowProgressIndicator(page),
+      "background-color"
+    );
+
+    expect(backgroundColorToken[0]).toBe("--colorsUtilityYin090");
+  });
+
+  test("background colour token should be correct when 'showCloseIcon' is true and the indicator is completed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <StepFlowComponent currentStep={2} totalSteps={3} showProgressIndicator />
+    );
+
+    const backgroundColorToken = await getDesignTokensByCssProperty(
+      page,
+      stepFlowProgressIndicator(page).nth(0),
+      "background-color"
+    );
+
+    expect(backgroundColorToken[0]).toBe("--colorsSemanticPositive500");
+  });
+
+  test("background colour should be correct when 'showCloseIcon' is true and the indicator is not completed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <StepFlowComponent currentStep={2} totalSteps={3} showProgressIndicator />
+    );
+
+    const backgroundColorToken = await getDesignTokensByCssProperty(
+      page,
+      stepFlowProgressIndicator(page).nth(2),
+      "background-color"
+    );
+
+    expect(backgroundColorToken[0]).toBe("--colorsActionDisabled600");
+  });
+});
+
+test.describe("Event checks for StepFlow component", () => {
+  test("should call onDismiss callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <StepFlowComponent
+        showCloseIcon
+        onDismiss={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    await stepFlowDismissIcon(page).click();
+    expect(callbackCount).toBe(1);
```

Comment:
**nitpick(non-blocking):** space here otherwise the QA's will break your legs.

```suggestion
    await stepFlowDismissIcon(page).click();
    
    expect(callbackCount).toBe(1);
```